lighthouse-core/test/audits/installable-manifest-test.js

```diff
@@ -85,97 +91,125 @@ describe('PWA: webapp install banner audit', () => {
+        assert.ok(items[0].reason.formattedDefault.includes('without a \'start_url\''));
       });
     });
 
     it('fails when a manifest contains no short_name', () => {
       const artifacts = generateMockArtifacts();
+      artifacts.InstallabilityErrors.errors.push({errorId: 'manifest-missing-name-or-short-name',
+        errorArguments: []});
       artifacts.WebAppManifest.value.short_name.value = undefined;
       const context = generateMockAuditContext();
 
       return InstallableManifestAudit.audit(artifacts, context).then(result => {
         assert.strictEqual(result.score, 0);
-        assert.ok(result.explanation.includes('short_name'), result.explanation);
-
-        const details = result.details.items[0];
-        assert.strictEqual(details.failures.length, 1, details.failures);
-        assert.strictEqual(details.hasStartUrl, true);
-        assert.strictEqual(details.hasShortName, false);
+        const items = result.details.items;
+        assert.ok(items[0].reason.formattedDefault.includes('does not contain a \'name\''));
       });
     });
 
     it('fails when a manifest contains no name', () => {
       const artifacts = generateMockArtifacts();
+      artifacts.InstallabilityErrors.errors.push({errorId: 'manifest-missing-name-or-short-name',
+        errorArguments: []});
       artifacts.WebAppManifest.value.name.value = undefined;
       const context = generateMockAuditContext();
 
       return InstallableManifestAudit.audit(artifacts, context).then(result => {
         assert.strictEqual(result.score, 0);
-        assert.ok(result.explanation.includes('name'), result.explanation);
-
-        const details = result.details.items[0];
-        assert.strictEqual(details.failures.length, 1, details.failures);
-        assert.strictEqual(details.hasStartUrl, true);
-        assert.strictEqual(details.hasName, false);
+        const items = result.details.items;
+        assert.ok(items[0].reason.formattedDefault.includes('does not contain a \'name\''));
       });
     });
 
     it('fails if page had no icons in the manifest', () => {
       const artifacts = generateMockArtifacts();
+      artifacts.InstallabilityErrors.errors.push({errorId: 'no-icon-available',
+        errorArguments: []});
       artifacts.WebAppManifest.value.icons.value = [];
       const context = generateMockAuditContext();
 
       return InstallableManifestAudit.audit(artifacts, context).then(result => {
         assert.strictEqual(result.score, 0);
-        assert.ok(result.explanation.includes('PNG icon'), result.explanation);
-
-        const details = result.details.items[0];
-        assert.strictEqual(details.failures.length, 2, details.failures);
-        assert.strictEqual(details.hasStartUrl, true);
-        assert.strictEqual(details.hasIconsAtLeast144px, false);
+        const items = result.details.items;
+        assert.ok(items[0].reason.formattedDefault.includes('icon was empty or corrupted'));
```

Comment:
```suggestion
        expect(items[0].reason).toBeDisplayString(/icon was empty or corrupted/);
```
or whatever it ends up being